src/MeetingsSDKAdapter.js

```diff
@@ -189,19 +199,35 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
    * @param {object} constraint Media stream constraints
    * @returns {Promise.<MediaStream>} Promise to requested media stream
    */
-  async getStream(ID, constraint) {
-    let localStream = null;
+  getStream(ID, constraint) {
+    return new Observable(async (subscriber) => {
+      let localStream = null;
 
-    try {
-      const sdkMeeting = this.fetchMeeting(ID);
+      try {
+        const sdkMeeting = this.fetchMeeting(ID);
 
-      [localStream] = await sdkMeeting.getMediaStreams(constraint);
-    } catch (error) {
-      // eslint-disable-next-line no-console
-      console.error(`Unable to retrieve local media stream for meeting "${ID}" with constraint "${constraint}"`, error);
-    }
+        subscriber.next({permission: 'ASKING', stream: localStream});
+        [localStream] = await sdkMeeting.getMediaStreams(constraint);
+        subscriber.next({permission: 'ALLOWED', stream: localStream});
+      } catch (error) {
+        let perm;
 
-    return localStream;
+        // eslint-disable-next-line no-console
+        console.error(`Unable to retrieve local media stream for meeting "${ID}" with constraint "${constraint}"`, error);
+
+        if (error instanceof DOMException && error.name === 'NotAllowedError') {
+          if (error.message === 'Permission dismissed') {
+            perm = 'DISMISSED';
+          } else {
+            perm = 'DENIED';
+          }
+        } else {
+          perm = 'ERROR';
+        }
+        subscriber.next({permission: perm, stream: localStream});
```

Comment:
you can use `null` instead of `localStream` here and move the declaration inside the `try` block:
```js
const [localStream] = await sdkMeeting.getMediaStreams(...);
```